modules/programs/nushell.nix

```diff
@@ -167,19 +180,35 @@ in {
     };
   };
 
-  config = mkIf cfg.enable {
+  config = lib.mkIf cfg.enable {
     home.packages = [ cfg.package ];
 
-    home.file = mkMerge [
+    home.file = lib.mkMerge [
       (let
         writeConfig = cfg.configFile != null || cfg.extraConfig != ""
-          || aliasesStr != "";
-
-        aliasesStr = concatStringsSep "\n"
-          (mapAttrsToList (k: v: "alias ${k} = ${v}") cfg.shellAliases);
-      in mkIf writeConfig {
-        "${configDir}/config.nu".text = mkMerge [
-          (mkIf (cfg.configFile != null) cfg.configFile.text)
+          || aliasesStr != "" || cfg.settings != { };
+
+        aliasesStr = lib.concatLines
+          (lib.mapAttrsToList (k: v: "alias ${k} = ${v}") cfg.shellAliases);
+      in lib.mkIf writeConfig {
+        "${configDir}/config.nu".text = lib.mkMerge [
+          (lib.mkIf (cfg.configFile != null) cfg.configFile.text)
+          (let
+            flattenSettings = settings:
+              let
+                unravel = prefixes: value:
+                  if (lib.isAttrs value && !isNushellInline value) then
+                    lib.flatten
+                    (map (key: unravel (prefixes ++ [ key ]) value.${key})
+                      (builtins.attrNames value))
+                  else
+                    lib.nameValuePair (lib.concatStringsSep "." prefixes) value;
+              in lib.listToAttrs (unravel [ ] settings);
+
+            flattenedSettings = flattenSettings cfg.settings;
+          in lib.mkIf (cfg.settings != { }) (lib.concatLines (lib.mapAttrsToList
+            (key: value: "$env.config.${key} = ${toNushell { } value}")
+            flattenedSettings)))
```

Comment:
An alternative implementation that I personally find slightly more readable and is potentially a bit more efficient since it avoids the intermediate attribute set (although I imagine you'll need many settings for it to be noticeable 🙂).

I haven't tested it beyond the test case and feel free to reject the suggestion if you prefer the original version:

```suggestion

          (let
            flattenSettings = settings:
              let
                joinDot = a: b: "${if a == "" then "" else "${a}."}${b}";
                unravel = prefix: value:
                  if lib.isAttrs value && !isNushellInline value then
                    lib.concatMap
                    (key: unravel (joinDot prefix key) value.${key})
                    (builtins.attrNames value)
                  else
                    [ (lib.nameValuePair prefix value) ];
              in unravel "" settings;

            mkLine = { name, value }: ''
              $env.config.${name} = ${toNushell { } value}
            '';

            settingsLines =
              lib.concatMapStrings mkLine (flattenSettings cfg.settings);
          in lib.mkIf (cfg.settings != { }) settingsLines)

```

Reply:
Ooh thanks! I prefer your implementation, it's _way_ easier to understand 😄